server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -100,6 +103,9 @@ public void execute(SearchContext searchContext) throws QueryPhaseExecutionExcep
                     new DocValueFormat[0]);
             return;
         }
+
+        LOGGER.trace("{}", new SearchContextRequestLog(searchContext));
```

Comment:
It might make sense to guard this log statement? (i.e. `if LOGGER.isTraceEnabled())`?)

Reply:
thanks